vertical-pod-autoscaler/pkg/recommender/model/cluster_test.go

```diff
@@ -82,6 +82,59 @@ func TestClusterGCAggregateContainerStateDeletesOld(t *testing.T) {
 	assert.Empty(t, vpa.aggregateContainerStates)
 }
 
+func TestClusterGCAggregateContainerStateDeletesOldEmpty(t *testing.T) {
+	// Create a pod with a single container.
+	cluster := NewClusterState()
+	vpa := addTestVpa(cluster)
+	addTestPod(cluster)
+
+	assert.NoError(t, cluster.AddOrUpdateContainer(testContainerID, testRequest))
+	// No usage samples added.
+
+	assert.NotEmpty(t, cluster.aggregateStateMap)
+	assert.NotEmpty(t, vpa.aggregateContainerStates)
+
+	assert.Len(t, cluster.aggregateStateMap, 1)
+	var creationTime time.Time
```

Comment:
cluster.aggregateStateMap[0].CreationTime = ...
instead of the loop? The test asserts the length is 1 right above this.

Reply:
The map keys are not ints but aggregateStateKeys. I could construct the key, but the fastest way is to iterate.

Reply:
Ack.